### PR TITLE
Add AmbExecutor.sol

### LINKED ISSUE #940: Prototype an extension for sending arbitrary transactions across the xDai AMB
One potentially useful use-case for Colony is for colonies to make decision on xDai, due to the low cost of gas, and then execute that decision on mainnet. Doing this would require leveraging the arbitrary message bridge, or AMB, which xDai has kindly made available.

We should explore the feasibility of this functionality. It will likely involve an extension which takes in arbitrary transactions (in the format we use for motions, etc) and invokes a function on the xDai AMB contract, to send the transaction to mainnet. One key open question is how gas costs are handled -- while there is a 2m gas limit per arbitrary message, xDai is almost certainly not taking responsibility for eth-side gas costs.

*Update*: After discussing with Auryn (who is developing similar functionality on the Gnosis Safe side), gas costs are paid by whoever executes the tx -- the idea being that validators submit signatures, but the actual execution can be triggered by anyone.

Some resources:

https://docs.tokenbridge.net/eth-xdai-amb-bridge/about-the-eth-xdai-amb

https://docs.tokenbridge.net/amb-bridge/how-to-develop-xchain-apps-by-amb

https://docs.tokenbridge.net/amb-bridge/gas-token-minting

https://github.com/gnosis/amb-module/blob/safebridge/contracts/SafeBridgeModule.sol

### PR DESCRIPTION
**Pull request by kronosapiens**
Closes #940 

## Notes

Encoding the AMB transaction:

`function requireToPassMessage(address _contract, bytes _data, uint256 _gas) external;`

`_contract` is the destination across the bridge.
`_data` is the transaction to execute, including the function signature and all encoded arguments.

On the AmbExecutor side, we have:

`function executeTransaction(address to, bytes memory action)`

Note that this entire function call (including the signature for `executeTransaction`, as well as `to` and `action`, are what we encoded in `_data` previously. Now, `to` is the address we are transacting to in order to perform our desired action, such as manipulating funds.

In other words:

```
_data = encode("executeTransaction", to, action);
```

## Resources

Based off https://github.com/gnosis/amb-module/blob/safebridge/contracts/SafeBridgeModule.sol

See also https://docs.tokenbridge.net/amb-bridge/how-to-develop-xchain-apps-by-amb#receive-a-method-call-from-the-amb-bridge

AMB tx dashboard: https://alm-rinkeby.herokuapp.com/100/0xeca054161cb422c35ded25b71926059162f0236bc35770e068c8ee984439ac78

### CODE REVIEW

**Comment by kronosapiens** [comment]
After more discussion with @jakzilla and @auryn-macmillan , we realized we didn't need this contract after all -- all AMB bridging functionality can be handled via arbitrary transactions or contracts developed elsewhere. Huzzah!
